src/serve/functions.ts

```diff
@@ -7,21 +7,32 @@ module.exports = {
   emulatorServer: undefined,
 
   async start(options: any): Promise<void> {
-    this.emulatorServer = new EmulatorServer(
-      new FunctionsEmulator(options, {
-        // When running the functions emulator through 'firebase serve' we disable some
-        // of the more adventurous features that could be breaking/unexpected behavior
-        // for those used to the legacy emulator.
-        disabledRuntimeFeatures: {
-          functions_config_helper: true,
-          network_filtering: true,
-          timeout: true,
-          memory_limiting: true,
-          protect_env: true,
-          admin_stubs: true,
-        },
-      })
-    );
+    const args: FunctionsEmulatorArgs = {
+      // When running the functions emulator through 'firebase serve' we disable some
+      // of the more adventurous features that could be breaking/unexpected behavior
+      // for those used to the legacy emulator.
+      disabledRuntimeFeatures: {
+        functions_config_helper: true,
+        network_filtering: true,
+        timeout: true,
+        memory_limiting: true,
+        protect_env: true,
+        admin_stubs: true,
+      },
+    };
+
+    // If hosting emulator is not being served but Functions is,
+    // we can use the port argument. Otherwise it goes to hosting and
+    // we use port + 1.
+    if (this.options.port) {
+      if (this.options.targets && this.options.targets.indexOf("hosting") < 0) {
+        args.port = this.options.port;
+      } else {
+        args.port = this.options.port + 1;
```

Comment:
Is there any chance of this colliding with the port used for another emulator if I did like `--only functions,firestore`?

Reply:
Only if you choose `8078` or `8079` basically and I think we should stop having people use `serve` for Firestore/RTDB anymore.

Reply:
Actually it turns out that `--only functions,firestore` never worked correctly (and still doesn't).  It starts Firestore only.  So no.